### PR TITLE
Alexis review

### PR DESCRIPTION
**Pull request by ametaireau**
Here are few changes I made while reviewing the code.

### CODE REVIEW

File: readinglist/__init__.py [hunk 1]
```diff
@@ -16,28 +13,34 @@
...
+# The API version is derivated from the module version.
+API_VERSION = 'v%s' % __version__.split('.')[0]
 
-def redirect_to_version(request):
-    """Redirect to the current version of the API."""
-    raise HTTPTemporaryRedirect(
-        '/%s/%s' % (API_VERSION, request.matchdict['path']))
 
+def handle_api_redirection(config):
+    """Adds a view which redirect to the current version of the API.
```
Comment (leplatrem) [inline readinglist/__init__.py]: Add (infinitive?)

File: readinglist/__init__.py [hunk 2]
```diff
@@ -48,25 +51,45 @@ def main(global_config, **settings):
...
+    handle_api_redirection(config)
+
+    config.route_prefix = '/%s' % API_VERSION
+
+    backend_module = config.maybe_dotted(settings['readinglist.backend'])
+    config.registry.backend = backend_module.load_from_config(config)
+
+    set_auth(config)
+
+    # Discover cornice super powers and include readinglist views.
```
Comment (leplatrem) [inline readinglist/__init__.py]: Include cornice and discover views ?

File: readinglist/authentication.py [hunk 3]
```diff
@@ -8,7 +8,10 @@
 
 
 def check_credentials(username, password, request):
-    """Basic auth backdoor. Used for testing (unit and loads)
+    """Basic auth backdoor.
+
+    This is mocked in the (unit and load) tests, and kept separate from the
+    authentication policies to ease the mocking.
```
Comment (leplatrem) [inline readinglist/authentication.py]: Hum, not really. Actually, it is not used in unit tests anymore. Just in loads tests. It is supposed to ease client development to bypass fxa integration...

File: readinglist/resource.py [hunk 4]
```diff
@@ -83,6 +83,18 @@ def wrapper(klass):
     return wrapper
 
 
+def decode_filter_value(value):
+    """Converts string value to native python values."""
```
Comment (leplatrem) [inline readinglist/resource.py]: Convert (infinitive for consistency)

File: readinglist/resource.py [hunk 5]
```diff
@@ -83,6 +83,18 @@ def wrapper(klass):
...
+    if value.lower() in ['on', 'true', 'yes', '1']:
+        value = True
+    elif value.lower() in ['off', 'false', 'no', '0']:
+        value = False
+    try:
+        return ast.literal_eval(value)
+    except ValueError:
+        return value
```
Comment (leplatrem) [inline readinglist/resource.py]: This could go to util.py, renamed to something like ``native_value()``

File: readinglist/resource.py [hunk 6]
```diff
@@ -140,16 +156,12 @@ def _extract_sorting(self, queryparams):
...
-        if value.lower() in ['on', 'true', 'yes', '1']:
-            value = True
-        elif value.lower() in ['off', 'false', 'no', '0']:
-            value = False
-        try:
-            return ast.literal_eval(value)
-        except ValueError:
-            return value
+    def merge_fields(self, stored, changes):
+        """Merge given request fields with stored fields"""
```
Comment (leplatrem) [inline readinglist/resource.py]: Merge changes into current record fields ?

Shouldn't we use ``self.record`` here instead of passing ``stored`` ?

**Comment by ametaireau** [comment]
updated.

File: readinglist/util.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import ast
```
Comment (Natim) [inline readinglist/util.py]: #24 will create a utils.py file, maybe we should rename this utils.py as well to share the same file.

**Comment by ametaireau** [comment]
I've also renamed util.py to utils.py. can we merge ?

**Pull request merged**